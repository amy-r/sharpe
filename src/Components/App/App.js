import React, { Component } from 'react';
// import logo from './logo.svg';
import './App.css';
import * as helper from '../../helper/helper.js'
import Table from '../Table/Table.js'

class App extends Component {
  constructor() {
    super();
    this.state = {
      currencyReturns: {}
    }
  }

  componentDidMount = async () => {
    const fetched = await helper.fetchApi('https://api.nomics.com/v0/sparkline')
    const currencyReturns = helper.getData(fetched)
    this.setState({ currencyReturns })
  }

  render() {

    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Welcome to Nomics</h1>
        </header>
        <Table currencyReturns={this.state.currencyReturns} />
      </div>
    );
  }
}

export default App;
