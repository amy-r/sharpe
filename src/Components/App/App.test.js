import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

it('welcomes you', () => {
  // const app = shallow(<App />);
  // expect(app.text()).toContain('Welcome to React');
})
