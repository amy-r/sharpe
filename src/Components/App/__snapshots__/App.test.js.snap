// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <h1
                    className="App-title"
          >
                    Welcome to Nomics
          </h1>
</header>,
        <Table
          currencyReturns={Object {}}
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="App-title"
>
            Welcome to Nomics
</h1>,
          "className": "App-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to Nomics",
            "className": "App-title",
          },
          "ref": null,
          "rendered": "Welcome to Nomics",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "currencyReturns": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <h1
                        className="App-title"
            >
                        Welcome to Nomics
            </h1>
</header>,
          <Table
            currencyReturns={Object {}}
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="App-title"
>
              Welcome to Nomics
</h1>,
            "className": "App-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to Nomics",
              "className": "App-title",
            },
            "ref": null,
            "rendered": "Welcome to Nomics",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "currencyReturns": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
